examples/complex-scene/src/navigation.rs

```diff
@@ -0,0 +1,112 @@
+use hotham::{
+    components::LocalTransform,
+    contexts::InputContext,
+    glam::{Affine3A, Vec3},
+    hecs::{self, World},
+    Engine,
+};
+
+use crate::State;
+
+/// Navigation system
+/// Allows the player to navigate by manipulating the stage transform
+pub fn navigation_system(engine: &mut Engine, state: &mut State) {
+    let world = &mut engine.world;
+    let input_context = &mut engine.input_context;
+    navigation_system_inner(world, input_context, engine.stage_entity, state);
+}
+
+fn navigation_system_inner(
+    world: &mut World,
+    input_context: &mut InputContext,
+    stage_entity: hecs::Entity,
+    state: &mut State,
+) {
+    // Get the stage transform.
+    let mut stage_transform = world.get::<&mut LocalTransform>(stage_entity).unwrap();
+    let global_from_stage = stage_transform.to_affine();
+
+    // Get the hand transforms.
+    let stage_from_left_grip = input_context.left.stage_from_grip();
```

Comment:
nit: Since all these branches are updating state, you could add a method to `State` that takes `InputContext` and updates it